linux/system/selinux.sls

```diff
@@ -10,21 +10,29 @@ include:
 
 selinux_config:
   cmd.run:
-  - names:
-    - "sed -i 's/enforcing/disabled/g' /etc/selinux/config; setenforce 0"
-    - "sed -i 's/permissive/disabled/g' /etc/selinux/config; setenforce 0"
-  - unless: cat '/etc/selinux/config' | grep 'SELINUX=disabled'
+  - name: "sed -i 's/SELINUX=[a-z][a-z]*$/SELINUX={{ system.selinux }}/' /etc/selinux/config"
+  - unless: cat '/etc/selinux/config' | grep 'SELINUX={{ system.selinux }}'
+  - require:
+    - pkg: linux_repo_prereq_pkgs
+
+permisive:
+  selinux.mode
 
 {%- else %}
 
 selinux_config:
-  selinux.mode:
-  - name: {{ system.get('selinux', 'permissive') }}
+  cmd.run:
+  - name: "sed -i 's/SELINUX=[a-z][a-z]*$/SELINUX={{ system.selinux }}/' /etc/selinux/config"
+  - unless: cat '/etc/selinux/config' | grep 'SELINUX={{ system.selinux }}'
```

Comment:
I think you could also reduce the code as it's duplicated for disabled vs. anything else action. Use the same states but decide on mode eg.:
```
{%- if system.selinux == 'disabled' %}
  {%- set mode = 'permisive' %}
{%- else %}
  {%- set mode = system.selinux %}
{%- endif %}
```